src/inference/tempering.jl

```diff
@@ -0,0 +1,178 @@
+###########################
+#   SAMPLING
+###########################
+
+"""
+Upon calling sample on a `TemperedSampler`, we must manually instantiate a `Sampler`
+to insert into the `TemperedSampler`, rather than wrapping the `TemperedSampler`
+itself in a `Sampler`
+"""
+function AbstractMCMC.sample(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    sampler::MCMCTempering.TemperedSampler{<:InferenceAlgorithm},
+    N::Integer;
+    chain_type=MCMCChains.Chains,
+    resume_from=nothing,
+    progress=PROGRESS[],
+    kwargs...
+)
+    sampler = MCMCTempering.TemperedSampler(
+        Sampler(sampler.internal_sampler, model),
+        sampler.Δ,
+        sampler.Δ_init,
+        sampler.N_swap,
+        sampler.swap_strategy
+    )
+
+    if resume_from === nothing
+        return AbstractMCMC.mcmcsample(rng, model, sampler, N;
+                                       chain_type=chain_type, progress=progress, kwargs...)
+    else
+        return resume(resume_from, N; chain_type=chain_type, progress=progress, kwargs...)
+    end
+end
+
+
+function AbstractMCMC.sample(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    sampler::MCMCTempering.TemperedSampler{<:InferenceAlgorithm},
+    ensemble::AbstractMCMC.AbstractMCMCEnsemble,
+    N::Integer,
+    n_chains::Integer;
+    chain_type=MCMCChains.Chains,
+    progress=PROGRESS[],
+    kwargs...
+)
+    sampler = MCMCTempering.TemperedSampler(
+        Sampler(sampler.internal_sampler, model),
+        sampler.Δ,
+        sampler.Δ_init,
+        sampler.N_swap,
+        sampler.swap_strategy
+    )
+
+    return AbstractMCMC.mcmcsample(rng, model, sampler, ensemble, N, n_chains;
+                                   chain_type=chain_type, progress=progress, kwargs...)
+end
+
+
+function AbstractMCMC.bundle_samples(
+    ts::Vector,
+    model::DynamicPPL.Model,
+    spl::MCMCTempering.TemperedSampler,
+    state::MCMCTempering.TemperedState,
+    chain_type::Union{Type{MCMCChains.Chains},Type{Vector{NamedTuple}}};
+    kwargs...
+)
+    return AbstractMCMC.bundle_samples(ts, model, spl.internal_sampler, state, chain_type; kwargs...)
+end
+
+
+
+###########################
+#   MODEL
+###########################
+
+struct TemperedEval{T<:AbstractFloat}
+    model :: DynamicPPL.Model
+    β     :: T
+end
+
+function (te::TemperedEval)(
+    rng::Random.AbstractRNG,
+    model::DynamicPPL.Model,
+    varinfo::DynamicPPL.AbstractVarInfo,
+    sampler::DynamicPPL.AbstractSampler,
+    context::DynamicPPL.AbstractContext,
+    args...
+)
+    context = DynamicPPL.MiniBatchContext(
```

Comment:
A bit clearer if one would use a different name here:
```suggestion
    minibatchcontext = DynamicPPL.MiniBatchContext(
```